openmc/mgxs/library.py

```diff
@@ -223,6 +229,28 @@ def domain_type(self, domain_type):
         cv.check_value('domain type', domain_type, tuple(openmc.mgxs.DOMAIN_TYPES))
         self._domain_type = domain_type
 
+    @domains.setter
+    def domains(self, domains):
+
+        # Use all materials, cells or universes in the geometry as domains
+        if domains == 'all':
+            self._domains = domains
+
+        # User specified a list of material, cell or universe domains
+        else:
+            if self.domain_type == 'material':
+                cv.check_iterable_type('domain', domains, openmc.Material)
+            elif self.domain_type in ['cell', 'distribcell']:
+                cv.check_iterable_type('domain', domains, openmc.Cell)
+            elif self.domain_type == 'universe':
+                cv.check_iterable_type('domain', domains, openmc.Universe)
+            else:
+                msg = 'Unable to set domains with ' \
+                      'domain type "{}"'.format(self.domain_type)
+                raise ValueError(msg)
+
+            self._domains = domains
```

Comment:
Is there an easy way to check the provided domain for validity? (i.e., I gave you cell 0, but there is no cell 0)